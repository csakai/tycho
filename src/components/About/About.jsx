import React from 'react';
import ReactMarkdown from 'react-markdown';
import PropTypes from 'prop-types';

export default class About extends React.Component {

    static propTypes = {
        text: PropTypes.string.isRequired
    }

    render() {
        return (
            <div className="about">
                <ReactMarkdown source={this.props.text} />
            </div>
        )
    }
}