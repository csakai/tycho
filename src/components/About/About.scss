@import 'globals.scss';

.about {
    overflow: auto;
    line-height: $font-line-height;
    flex: 1;

    &::-webkit-scrollbar {
        width: 5px;

        &-thumb {
            border-radius: 2px;
            background-color: transparent;
        }
    }

    &:hover {
        &::-webkit-scrollbar-thumb {
            background-color: #808080;
        }
    }
}