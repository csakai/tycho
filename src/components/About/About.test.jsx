import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import About from './About';

describe('About Box Component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<About text="Hello, world!" />);
  });

  it('should render the about box successfully', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});