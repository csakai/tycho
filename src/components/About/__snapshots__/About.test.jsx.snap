// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Box Component should render the about box successfully 1`] = `
<div
  className="about"
>
  <ReactMarkdown
    containerTagName="div"
    parserOptions={Object {}}
    source="Hello, world!"
  />
</div>
`;
